source/features/action-pr-link.tsx

```diff
@@ -0,0 +1,34 @@
+import * as pageDetect from 'github-url-detection';
+import select from 'select-dom';
+
+import features from '../feature-manager.js';
+import observe from '../helpers/selector-observer.js';
+
+async function add(row: HTMLDivElement): Promise<void> {
+	const prNumber = select('a[data-hovercard-type="pull_request"]', row)?.textContent?.slice(1);
+
+	if (prNumber === undefined) {
+		return;
+	}
+
+	select('a.Link--primary', row)!.href += `?pr=${prNumber}`;
+}
+
+async function init(signal: AbortSignal): Promise<void> {
+	observe('div.Box-row[id^=check_suite_]', add, {signal});
```

Comment:
Just select the `a[data-hovercard-type="pull_request"]` instead, so you don't have to `select` it again and then `return` when undefined.

```suggestion
	observe('div.Box-row[id^=check_suite_] a[data-hovercard-type="pull_request"]', add, {signal});
```

and then probably:

```ts
function add(prLink: HTMLAnchorElement) {
	const runLink = prLink.closest('.Row')!.querySelector('a.Link--primary')!;
	const parameters = new URLSearchParams(runLink.search);
	parameters.set('pr', prNumber);
	runLink.search = String(parameters);
}
```